static/js/util/util.js

```diff
@@ -95,3 +95,31 @@ export function makeCourseStatusDisplay(course, now = moment()) {
     return "";
   }
 }
+
+/* eslint-disable camelcase */
+/**
+ * Validates the profile
+ *
+ * @param {Object} profile The user profile
+ * @returns {Object} Validation errors or an empty object if no errors
+ */
+export function validateProfile(profile) {
+  let errors = {};
+
+  let required = {
+    'first_name': "First name",
+    'last_name': "Last name",
+    'preferred_name': "Preferred name",
+    'gender': "Gender",
+    'preferred_language': "Preferred language"
```

Comment:
these don't match the things we display in the UI (i.e. 'First Name' is displayed in the UI as 'Given Name').

might be good to make them match, it's a little incongruous:

![name_mismatch](https://cloud.githubusercontent.com/assets/6207644/14690357/cd3f0a5e-0719-11e6-977e-a7262083cd47.png)